Source/DafnyTestGeneration/ProgramModifier.cs

```diff
@@ -119,7 +120,7 @@ public abstract class ProgramModifier {
         }
       }
       toRemove.ForEach(x => program.RemoveTopLevelDeclaration(x));
-      return Utils.DeepCloneProgramAndReresolve(program, DafnyOptions.O);
+      return Utils.DeepCloneResolvedProgram(program, options);
```

Comment:
This seems like a potentially substantive change. Are you making this change because `DeepCloseResolvedProgram` didn't exist before but is more efficient now that it does exist? Or is there another reason?